[Security GenAI][Integration Assistant] Create OpenAPI spec for Integration Assistant APIs

bhapas opened the pull request:
## Summary

This PR adds OpenAPI spec to the integration assistant APIs introduced in https://github.com/elastic/kibana/pull/184296

---

## Follow-up changes

Had a discussion with @maximpn and realized that currently `zod library` does not support recursion and a [workaround](https://github.com/colinhacks/zod?tab=readme-ov-file#recursive-types) is followed. 

Hence the `xpack-codegen-enabled` is set to `false` and the generated `TS` files are renamed to remove `.gen` from the naming to keep them static until a [fix is in place](https://github.com/elastic/kibana/pull/186221).


### Checklist

Delete any items that are not applicable to this PR.


- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)

### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

bhapas committed:
Create OpenAPI spec for Integration Assistant APIs

bhapas committed:
Merge branch 'main' into openai_integration_assistant

bhapas committed:
Merge remote-tracking branch 'upstream/main' into openai_integration_assistant

bhapas committed:
Merge branch 'main' into openai_integration_assistant

bhapas committed:
Replace types with zod generated OpenAPI types

bhapas commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into openai_integration_assistant

bhapas commented:
/ci

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

bhapas commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into openai_integration_assistant

kibanamachine committed:
Merge branch 'main' into openai_integration_assistant

bhapas commented:
/ci

bhapas committed:
fix types

bhapas committed:
Merge remote-tracking branch 'upstream/main' into openai_integration_assistant

bhapas commented:
/ci

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

bhapas committed:
Merge remote-tracking branch 'upstream/main' into openai_integration_assistant

bhapas commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into openai_integration_assistant

bhapas commented:
/ci

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

maximpn commented on x-pack/plugins/integration_assistant/server/routes/categorization_routes.ts:
```diff
@@ -36,64 +39,53 @@ export function registerCategorizationRoutes(
-        try {
-          results = (await graph.invoke({
+          const graph = await getCategorizationGraph(client, model);
+          const results: CategorizationResponse = await graph.invoke({
```
To be 100% sure about the return type you could validate the response like

```ts
const results = await graph.invoke({
  packageName,
  dataStreamName,
  datastreamName,
  rawSamples,
  currentPipeline,
}));

return res.ok({ body: CategorizationApiResponse.parse(results) });
```

bhapas replied:
Changed in https://github.com/elastic/kibana/pull/186085/commits/ceb8d3e560a914767431617127638289d6e0a3ad

maximpn reviewed (approved):
I verified only added OpenAPI specs semantics, corresponding Zod schemas and TS types and how it's used since I'm not deeply familiar with the domain OpenAPI specs were added to.

Added  OpenAPI specs semantics, corresponding Zod schemas and TS types look good to me.

Ideally someone familiar with the domain should double check it for example @P1llus as the plugin author.

bhapas commented:
>Ideally someone familiar with the domain should double check it for example @P1llus as the plugin author.

@maximpn Thanks for the review. @P1llus is on leave and I have taken over the plugin development and we are on same team. I would still like @semd to do a review as he is the implementing the UI parts of this plugin.

bhapas commented on x-pack/plugins/integration_assistant/common/api/build_integration/build_integration.schema.yaml:
```diff
@@ -0,0 +1,33 @@
+openapi: 3.0.0
```
@maximpn Should we use `3.0.3` here?

maximpn replied:
Technically it should be `3.0.3` but the majority of specs in Kibana use  `3.0.0`. Since it's a separate plugin it's ok to specify `3.0.3`. It should be transparent for the tooling but keep an eye on it just in case.

bhapas replied:
Fixed in https://github.com/elastic/kibana/pull/186085/commits/ceb8d3e560a914767431617127638289d6e0a3ad

bhapas committed:
Fix openapi version

bhapas commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into openai_integration_assistant

MiriamAparicio committed:
[ObsUX][Infra] Fix container metrics not showing correct fields for docker (#186363)

## Summary

Some conditions were not correct on the render of the KPI, this didn’t
show in the e2e tests because with synthtrace we just generate data for
docker without kubernetes fields so it will be always showing docker
fields. Apparently is possible to have both docker and k8s fields
(kubernetes.container & docker.container), at least on the test data we
have on the oblt-clusters.
What we did is check if docker.container or kubernetes.container exist,
in case of none, charts will not display, in case of both we show docker
metric charts, otherwise k8s metrics charts

For the metrics section the condition was correct but not for the KPIs

hop-dev committed:
[Entity Analytics] Flaky Cypress test fixes (#186081)

### Fix 1: Press refresh button after using date picker on dashboard

Closes https://github.com/elastic/kibana/issues/183662
Closes https://github.com/elastic/kibana/issues/183635
Closes https://github.com/elastic/kibana/issues/183982

We shouldnt have to do this, but there is some condition where clicking
apply gets the UI in a stuck state and I cant figure it out.

### Fix 2: Use more specific selector for search bar on risk score
preview page

Closes https://github.com/elastic/kibana/issues/184133

We were using a generic selector `[data-test-subj="unifiedQueryInput"]
textarea` which was bringing back two elements sometimes (no idea why),
I have given the box its own dedicated selector to hopefully prevent
this from happening

---------

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

semd commented on x-pack/plugins/integration_assistant/server/routes/ecs_routes.ts:
```diff
@@ -34,70 +33,60 @@ export function registerEcsRoutes(router: IRouter<IntegrationAssistantRouteHandl
               rawSamples,
-            })) as EcsMappingApiResponse;
+            });
+          return res.ok({ body: results });
```
NIT: Should we `EcsMappingResponse.parse` this one too?

semd reviewed (approved):
LGTM!

semd commented on x-pack/plugins/integration_assistant/server/integration_builder/dev_folders.ts:
```diff
@@ -5,10 +5,10 @@
-import { join as joinPath } from 'path';
 import nunjucks from 'nunjucks';
+import { join as joinPath } from 'path';
 import type { Integration } from '../../common';
-import { ensureDirSync, createSync } from '../util';
+import { createSync, ensureDirSync } from '../util';
```
This is not making any difference, could we rollback this change?

semd commented on x-pack/plugins/integration_assistant/server/integration_builder/pipeline.ts:
```diff
@@ -4,8 +4,8 @@
  */
-import { join as joinPath } from 'path';
 import yaml from 'js-yaml';
+import { join as joinPath } from 'path';
```
Same, can this be rolled back?

ersin-erdal committed:
Add connector and actionTaskParams modelVersions (#184542)

Resolves: #177059

This PR adds model versions for `action` and `action_task_params` saved
objects.
I also re-organised the schema and model version file structure in
alerting, task_manager and actions plugins.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

nikitaindik committed:
[Security Solution] `DetectionRulesClient`: return `RuleResponse` from `createCustomRule` and `createPrebuiltRule` (#185748)

**Partially addresses: https://github.com/elastic/kibana/issues/184364**

## Summary
This PR changes `createCustomRule` and `createPrebuiltRule` methods to
return `RuleResponse` instead of `RuleAlertType`.

This is a continuation of the effort to improve `DetectionRulesClient`.
As a part of it we want to make the client return `RuleResponse` from
all its public methods. This is good because it lets us hide rule's
internal structure and centralise conversions in one place – in the
client. So in this and upcoming PRs we are going to convert
`RuleAlertType` to `RuleResponse` within the client's methods and return
the `RuleResponse` object.

doakalexi committed:
[ResponseOps] ES|QL rule type improvements - write query results to the alert doc (#184541)

Resolves https://github.com/elastic/response-ops-team/issues/200

## Summary

This PR copies the fields from the ES|QL query results to the alert doc.
Now that we are writing everything to the alert doc I removed the source
fields selector from the ES|QL ui.


### Checklist

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios


### To verify

- Create an ES Query ES|QL rule.
- Go to [dev tools](http://localhost:5601/app/dev_tools#/console) and
run the query below to verify that the query results are written to the
alert doc.
```
GET .internal.alerts-stack.alerts-default*/_search
```

jgowdyelastic committed:
[ML] Refresh jobs list after import (#184757)

Fixes https://github.com/elastic/kibana/issues/184063

Adds a callback prop to allow the import flyout to refresh the jobs list
after importing jobs or synchronising saved objects.

lcawl committed:
[DOCS] Deprecate old advanced settings (#186309)

jloleysens committed:
[OAS] Deprecations (#186062)

## Summary

Adds support for the [`deprecated`
field](https://swagger.io/specification/) at the operation level to our
router conversion logic.

Close https://github.com/elastic/kibana/issues/186003

joemcelroy committed:
[Search] [Playground] Semantic text support (#186268)

## Summary

This adds semantic_text support to Chat playground. 

- We detect the presence of semantic_text and favour these fields over
any other ones.
- We do a nested query if the field is a semantic_text type, and use the
inner_hits to populate the context
- This was manually tested through the QA scenarios list and I added a
couple more scenarios for semantic_text.
- Added to telemetry key with `_SEMANTIC` entry

Coverage is still units but this change is pretty well covered. FTRs
will follow once we have basic "full chat" flow.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

---------

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

maximpn committed:
[Security Solution] Add missing alerts (signals) API endpoints OpenAPI specs (#184838)

**Addresses:** https://github.com/elastic/kibana/issues/183661

## Summary

This PR adds missing OpenAPI specs for the following API endpoints (Alerts API and Alerts Migration API) available in both Serverless and ESS

- `POST /api/detection_engine/signals/status`
- `POST /api/detection_engine/signals/tags`
- `POST /api/detection_engine/signals/search`

and API endpoints available only in ESS

- `POST /api/detection_engine/signals/migration_status`
- `POST /api/detection_engine/signals/migration`
- `POST /api/detection_engine/signals/finalize_migration`
- `DELETE /api/detection_engine/signals/migration`

**Note:** Code generation is enabled for the added specs to verify that it works and produces expected results. Generated Zod schemas and types aren't integrated in the route's code.

justinkambic committed:
[Synthetics] Fix TLS cert view for > 3 monitors per cert (#186204)

## Summary

Resolves #183985.

This patch makes it so we can see more than three monitors per cert in
the TLS page of Synthetics.

I've set the default limit to the round number of 100. Realistically,
users could have more than 100 monitors for the same cert, but also
realistically, this page would need a layout overhaul to display that
many monitors in a single row as well.

### Before

<img width="1414" alt="image"
src="https://github.com/elastic/kibana/assets/18429259/26357eee-b909-407a-af61-a61b5033e899">


### After

<img width="1412" alt="image"
src="https://github.com/elastic/kibana/assets/18429259/51f47e4e-7b5b-4fa6-b20c-90344c0c6912">


## Testing this PR

Follow the simple repro instructions in the linked issue. On `main`, you
should only see 3 or less monitors. On this branch, you should see all
monitors.

tkajtoch committed:
Upgrade EUI to v95.0.0 (#185943)

`v94.6.0` ⏩ `v95.0.0-backport.0`

_[Questions? Please see our Kibana upgrade
FAQ.](https://github.com/elastic/eui/blob/main/wiki/eui-team-processes/upgrading-kibana.md#faq-for-kibana-teams)_

---

##
[`v95.0.0-backport.0`](https://github.com/elastic/eui/releases/v95.0.0-backport.0)

**This is a backport release only intended for use by Kibana.**

- Updated `EuiSteps` to support a new `titleSize="xxs"` style, which
outputs the same title font size but smaller unnumbered step indicators
([#7813](https://github.com/elastic/eui/pull/7813))
- Updated `EuiStepsHorizontal` to support a new `size="xs"` style, which
outputs smaller unnumbered step indicators
([#7813](https://github.com/elastic/eui/pull/7813))
- Updated `EuiStepNumber` to support new `titleSize="none"` which omits
rendering step numbers, and will only render icons
([#7813](https://github.com/elastic/eui/pull/7813))

## [`v95.0.0`](https://github.com/elastic/eui/releases/v95.0.0)

- Added `move` glyph to `EuiIcon`
([#7789](https://github.com/elastic/eui/pull/7789))
- Updated `EuiBasicTable` and `EuiInMemoryTable`s with `selection` - the
header row checkbox will now render an indeterminate state if some (but
not all) rows are selected
([#7817](https://github.com/elastic/eui/pull/7817))

**Bug fixes**

- Fixed an `EuiDataGrid` visual bug when using `lineCount` row heights
where the clamped text was still visible for some font sizes
([#7793](https://github.com/elastic/eui/pull/7793))
- Fixed `EuiSearchBar`'s filter configs to always respect `autoClose:
false` ([#7806](https://github.com/elastic/eui/pull/7806))

**Breaking changes**

- Removed deprecated `EUI_CHARTS_THEME_DARK`, `EUI_CHARTS_THEME_LIGHT`
and `EUI_SPARKLINE_THEME_PARTIAL` exports
([#7682](https://github.com/elastic/eui/pull/7682))
- Removed deprecated `euiPalettePositive` and `euiPaletteNegative`. Use
`euiPaletteGreen` and `euiPaletteRed` instead
([#7808](https://github.com/elastic/eui/pull/7808))
- Removed `type="inList"` from `EuiCheckbox`. Simply omit passing a
`label` prop to render this style of checkbox
([#7814](https://github.com/elastic/eui/pull/7814))
- Removed the unused `compressed` prop from `EuiCheckbox` and
`EuiRadio`. This prop was not doing anything on individual components.
([#7818](https://github.com/elastic/eui/pull/7818))

**CSS-in-JS conversions**

- Converted `EuiCheckboxGroup` to Emotion
([#7818](https://github.com/elastic/eui/pull/7818))
- Converted `EuiRadioGroup` to Emotion
([#7818](https://github.com/elastic/eui/pull/7818))

---------

Co-authored-by: Cee Chen <constance.chen@elastic.co>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Cee Chen <549407+cee-chen@users.noreply.github.com>

eokoneyo committed:
Improve logic for dashboard duplication title computation (#185056)

## Summary

follow up to
https://github.com/elastic/kibana/pull/184777#discussion_r1631353032

Previously, if a user has cloned a dashboard N times, in the worst case
that the next duplication attempt is initiated from the initial
dashboard N network calls would be required to find a unique title for
the next uniquely titled clone. This update short circuits that process
by querying for the last created dashboard, getting it's duplicationId,
increment it and proposing it as the next dashboard duplicate title,
with this approach there's still a chance we might suggest a dashboard
title that's been claimed if the user modifies the dashboard title to a
higher duplication Id after creation, especially that we are querying
for the last created and this is situation we can't correct for, the
resolution for the appropriate title in that case would be on the user.

## How to test
- Create couple of duplicated dashboards, the titles should follow the
normal increment progression ie. n+1
- Create some that skip the expected pattern by any increment of choice,
on attempting to duplicate this dashboard the suggested title should be
previous + 1
- Choose to duplicate any previously created dashboard the suggested
title should be the increment of the last duplicated dashboard


### Checklist


<!-- Delete any items that are not applicable to this PR. -->

<!--
- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials -->
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
<!-- 
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to
identify risks that should be tested prior to the change/feature
release.

When forming the risk matrix, consider some of the following examples
and how they may potentially impact the change:

| Risk | Probability | Severity | Mitigation/Notes |

|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space.
| Low | High | Integration tests will verify that all features are still
supported in non-default Kibana Space and when user switches between
spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions
when multiple Kibana nodes are polling for the same tasks. | High | Low
| Tasks are idempotent, so executing them multiple times will not result
in logical error, but will degrade performance. To test for this case we
add plenty of unit tests around this logic and document manual testing
procedure. |
| Code should gracefully handle cases when feature X or plugin Y are
disabled. | Medium | High | Unit tests will verify that any feature flag
or plugin combination still results in our service operational. |
| [See more potential risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
-->

---------

Co-authored-by: Catherine Liu <catherine.liu@elastic.co>

vitaliidm committed:
[Security Solution][Detection Engine] fixes and unskips new terms FTR tests (#186283)

## Summary

- addresses https://github.com/elastic/kibana/issues/180236
- replaces kibana expect with jest expect

kdelemme committed:
feat(slo): Improve permission check on SLO pages (#182609)

lcawl committed:
[HTTP/OAS] Add tags and descriptions for uptime setting APIs (#184268)

nchaulet committed:
[Fleet] Add global data tags telemetry (#186390)

jbudz committed:
[ci] Remove 8.13 (#186377)

8.14 is released.  8.13 tests do not need to run now.

gsoldevila committed:
Remove IE check + banner (#186396)

## Summary

Address https://github.com/elastic/kibana/issues/138089

lcawl committed:
[OAS] Edits OpenAPI documents for machine learning sync API (#186318)

Dosant committed:
Add dashboard metadata to the info flyout (#185941)

## Summary

Close https://github.com/elastic/kibana-team/issues/898

- Show createdAt, createdBy, updatedAt, updatedBy in info flyout. Add a
bit of special handling for managed objects and when info is not
available.
- I had to extract some components into a separate package to use them
in contentEditor package
- tiny tweaks to column width and "no creator" state 



![Screenshot 2024-06-12 at 17 01
45](https://github.com/elastic/kibana/assets/7784120/b2093c03-67a0-49a5-8a45-93d9e57813ca)

**Unknown creator:**


![Screenshot 2024-06-12 at 17 01
53](https://github.com/elastic/kibana/assets/7784120/3e520f6a-9a19-455f-b564-571c3ad81b16)

**For managed objects:**

![Screenshot 2024-06-12 at 17 01
57](https://github.com/elastic/kibana/assets/7784120/36ce1465-09a4-4936-a9f1-ca5794d45a7a)

**Just created, no updates yet**

![Screenshot 2024-06-12 at 17 02
18](https://github.com/elastic/kibana/assets/7784120/1431210e-ce83-4409-ab99-6184b6f87d3a)

bhapas committed:
fix pr comments

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

bhapas committed:
Merge remote-tracking branch 'upstream/main' into openai_integration_assistant
